cards.js

```diff
@@ -1,23 +1,69 @@
-var values  = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"];
-var suits   = ["Clubs", "Diamonds", "Hearts", "Spades"];
-
-function shuffle(array){
-  var currentIndex = array.length;
-  var randomIndex;
-  var temporaryValue;
-  while(currentIndex > 0){
-    randomIndex   = Math.floor(Math.random() * currentIndex);
-    currentIndex -= 1;
-
-    temporaryValue       = array[currentIndex];
-    array[currentIndex]  = array[randomIndex];
-    array[randomIndex]   = temporaryValue;
+  var values  = [2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K", "A"]; // all the values in the deck
+  var suits   = ["Clubs", "Diamonds", "Hearts", "Spades"]; // all the suits in the deck
+  var deck = []
```

Comment:
It's bad practice to declare a `deck` variable. Since the variable is outside of the `build_deck` function, if you call `build_deck` multiple times, you'll keep adding more and more cards to this array. Another way to put this is your `build_deck` function is broken. If you call it a second time, you'll get a deck with 104 cards, instead of 52.